### DISCUSSION TITLE
PUT Method Not Updating Groups via Keycloak REST API

### DISCUSSION THREAD

**Discussion by flaviodotcom** [keycloak/keycloak#35528] [Q&A]
Hello everyone,

I've encountered an issue with the Keycloak REST API when working with user groups. While the POST method successfully creates groups, the PUT method does not update the groups as expected.

According to the official documentation for the `REST API`, the entire `UserRepresentation` object should be passed in the request body when using the PUT method to update a user. However, it appears that the group information in the UserRepresentation is not being considered or applied during the update process.

![image](https://github.com/user-attachments/assets/8f89f19c-fa79-4915-9187-8bd2662bf64d)

This seems to be a potential issue in the UserResource class, which may not be handling the group updates correctly. Could you confirm if this is a bug or the expected behavior? If it's a bug, I’d be happy to assist with additional details or testing to help resolve it.

**Comment by Nitin-iotina** [comment]
I am also getting the same behavior on keycloak:23.0.6. Did you find any solution?
Request:

`
curl --location --request PUT 'https://*******.com/admin/realms/realm1/users/e99cafcd-639b-4631-bf45-e7e64785020f' \--header 'Content-Type: application/json' \--header 'Authorization: ••••••' \--data '{    "firstName" : "Test",    "groups": [ "Group", "Group1", "Group2"]}'`
